frontend/src/ui/Floating.tsx

```diff
@@ -226,7 +231,9 @@ export const FloatingTrigger: React.FC<FloatingTriggerProps> = ({ children }) =>
 
     return React.cloneElement(children, {
         "data-floating-state": context.open ? "open" : "closed",
-        ...context.getReferenceProps({ ref: context.refs.reference, ...children.props }),
+        ...context.getReferenceProps({ ref: context.refs.reference,
+            onClick: () => context.setOpen && context.setOpen(false),
+            ...children.props }),
```

Comment:
```suggestion
        ...context.getReferenceProps({ 
            ref: context.refs.reference,
            onClick: () => context.setOpen?.(false),
            ...children.props,
        }),
```

Reply:
aah I tried that but was missing the dot after the qustion mark